File: bench/io/bench_write.js
// Write a million formatted lines. See bench_write.c for what this measures.
import * as std from "std";
import * as os from "os";

const LINES = 1000000;
const t0 = Date.now();
const f = std.open("io_out_qjs.txt", "wb");
for (let i = 0; i < LINES; i++) f.puts(i + " payload " + ((i * 7) % 9973) + "\n");
f.close();
const ms = Date.now() - t0;

const st = os.stat("io_out_qjs.txt")[0];
os.remove("io_out_qjs.txt");
console.log("result: " + st.size);
console.log("elapsed: " + ms);

File: bench/io/bench_write.go
// Write a million formatted lines. See bench_write.c for what this measures.
package main

import (
	"bufio"
	"fmt"
	"os"
	"time"
)

const lines = 1000000

func main() {
	t0 := time.Now()
	f, err := os.Create("io_out_go.txt")
	if err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
	w := bufio.NewWriter(f)
	for i := 0; i < lines; i++ {
		fmt.Fprintf(w, "%d payload %d\n", i, i*7%9973)
	}
	w.Flush()
	f.Close()
	ms := time.Since(t0).Milliseconds()

	st, _ := os.Stat("io_out_go.txt")
	size := st.Size()
	os.Remove("io_out_go.txt")
	fmt.Printf("result: %d\n", size)
	fmt.Printf("elapsed: %d\n", ms)
}
